library/std/src/sys/unix/stack_overflow.rs

```diff
@@ -129,28 +129,36 @@ mod imp {
         drop_handler(MAIN_ALTSTACK.load(Ordering::Relaxed));
     }
 
-    unsafe fn get_stackp() -> *mut libc::c_void {
+    unsafe fn get_stack() -> libc::stack_t {
         // OpenBSD requires this flag for stack mapping
         // otherwise the said mapping will fail as a no-op on most systems
         // and has a different meaning on FreeBSD
         #[cfg(any(target_os = "openbsd", target_os = "netbsd", target_os = "linux",))]
         let flags = MAP_PRIVATE | MAP_ANON | libc::MAP_STACK;
```

Comment:
Tangentially, should this also include android?

Reply:
afaik it doesn't do anything on Linux, it's just defined, which is not the same thing.

Reply:
If we wanted to use it on everything where it's defined, we would also do Fuchsia:

https://github.com/rust-lang/libc/blob/9bfa899501d25f9eff56a0a67ed5b13a13126a23/src/fuchsia/mod.rs#L2936

And neutrino:
https://github.com/rust-lang/libc/blob/9bfa899501d25f9eff56a0a67ed5b13a13126a23/src/unix/nto/mod.rs#L2485

And also Dragonfly and FreeBSD.
However, our libc doesn't declare that value for Android. There's a lot of aberrations like that where something available on both Linux and Android isn't enabled for Android, unfortunately.